packages/ui5-tooling-transpile/lib/task.js

```diff
@@ -31,54 +32,150 @@ module.exports = async function ({ workspace, dependencies, /* taskUtil,*/ optio
 			) {
 				const filePath = resourcePath.replace(new RegExp("\\.[^.]+$"), ".js");
 
-				// read file into string
+				// read source file
 				const source = await resource.getString();
 
-				// transpile the source
-				config.debug && log.info(`Transpiling resource ${resource.getPath()}`);
-				const result = await babel.transformAsync(
-					source,
-					Object.assign({}, babelConfig, {
-						filename: resource.getPath() // necessary for source map <-> source assoc
-					})
-				);
+				// store the ts source code in the sources map
+				if (config.transpileTypeScript) {
+					sourcesMap[resourcePath] = source;
+				}
 
-				resource.setString(normalizeLineFeeds(result.code));
-				resource.setPath(filePath);
-				workspace.write(resource);
+				// we ignore d.ts files for transpiling
+				if (!resourcePath.endsWith(".d.ts")) {
+					// mark source for omit from build result
+					taskUtil.setTag(resource, taskUtil.STANDARD_TAGS.OmitFromBuildResult, true);
 
-				// create SourceMap resource if available
-				if (result.map) {
-					//add the transpiled file name
-					result.map.file = path.basename(filePath);
-					config.debug && log.info(`  + sourcemap ${filePath}.map`);
+					// transpile the source
+					config.debug && log.info(`Transpiling resource ${resourcePath}`);
+					const result = await babel.transformAsync(
+						source,
+						Object.assign({}, babelConfig, {
+							filename: resourcePath // necessary for source map <-> source assoc
+						})
+					);
 
-					const resourceMap = resourceFactory.createResource({
-						path: `${filePath}.map`,
-						string: JSON.stringify(result.map)
+					// create the ts file in the workspace
+					const transpiledResource = resourceFactory.createResource({
+						path: filePath,
+						string: normalizeLineFeeds(result.code)
 					});
+					workspace.write(transpiledResource);
+
+					// create sourcemap resource if available
+					if (result.map) {
+						result.map.file = path.basename(filePath);
+						config.debug && log.info(`  + sourcemap ${filePath}.map`);
 
-					workspace.write(resourceMap);
+						const resourceMap = resourceFactory.createResource({
+							path: `${filePath}.map`,
+							string: JSON.stringify(result.map)
+						});
+
+						workspace.write(resourceMap);
+					}
 				}
 			}
 		})
 	);
+
+	// generate the dts files for the ts files
+	if (config.transpileTypeScript && !config.skipGenerateDTS) {
+		try {
+			// dynamically require typescript
+			const ts = require("typescript");
+
+			// options to generate d.ts files only
+			const options = {
+				allowJs: true,
+				declaration: true,
+				emitDeclarationOnly: true
+				//traceResolution: true,
+			};
+
+			// Create a Program with an in-memory emit
+			const host = ts.createCompilerHost(options);
+			(host.getCurrentDirectory = function () {
+				return "";
+			}),
+				(host.fileExists = function (file) {
+					if (sourcesMap[file]) {
```

Comment:
can be shortened: 
```
return sourcesMap[file] ? true : fs.existsSync(file)
```

Reply:
Damn, the auto completion of VSCode completely changed the code and put it into brackets... FYI: the reason why I have chosen those lengthy code was to be able to put break points inside and add additional logging - therefore I prefer to stick to the lengthy code. And the reason why I used classic functions is to be able to use the `arguments` when logging... 😉